src/public/interfaces.ts

```diff
@@ -519,3 +519,75 @@ export interface FrameContext {
    */
   websiteUrl: string;
 }
+
+export interface Error {
```

Comment:
I kept description in this object, because I thought in certain scenarios if Teams App wants to give more insight into the error or suggestions to developer which would help developer in resolving the error, it can be done using description. 
For example, let's say web-app developer is invoking some API with wrong input parameters and Teams App returns an errorCode 4000 (Invalid arguments). The error code is generic and doesn't give details to developer what exactly is wrong with parameters he passed. 
So, if Teams App can pass additional info like "mandatory field mediaType is missing", it would help developer a lot in quickly identifying his mistake.

The other reason I defined Error interface rather than directly sending ErrorCode back to web-app is that passing error object is extensible and can be extended in future. 

Regarding the confusion of name conflict, could you please suggest appropriate name for error interface if you agree with the need of having an error interface

@anandvnath, @ydogandjiev - Waiting for your feedback on above point. Post that I will make the required change as per the suggestion.

Reply:
The scenario you described makes sense. One issue I see is about language. If we start passing description, from API,  SDK / native will be responsible for maintaining and translating these strings. So, it is better if we can avoid handling strings that can be shown to the user. Also this gives app a free hand on what they want to show. app1 and app2 may want different things in the same errorCode. This is why I do not like description.

I agree that having Error as an interface is more extensible. SdkError is an option for a name.

Reply:
In the documentation of description field, I wrote that this description is not for consumption of the end user of the web-app and it will NOT be localized. The description is meant only for developer of the web-app to assist him in resolving errors during web-app development. During run-time of web-app, what needs to be shown to end user, it is the responsibility of the web-app developer and he should do that based on error code.

Reply:
Ok. That would make sense.